src/main.rs

```diff
@@ -135,11 +144,66 @@ async fn main() -> Result<()> {
 
             futures::future::try_join_all(group_futures).await?;
         }
+
+        CommandLineArgs::GroupList { user_id } => {
+            let device_context = find_device_for_user(&user_id)?;
+            let sdk = ironoxide::initialize(&device_context, &Default::default()).await?;
+            let groups = sdk.group_list().await?;
+            let group_ids = groups
+                .result()
+                .iter()
+                .map(|meta_result| meta_result.id().id())
+                .collect::<Vec<_>>();
+            println!("Groups found: {:?}", group_ids)
+        }
+
+        CommandLineArgs::GroupAddMembers {
+            user_id,
+            group_id,
+            new_members,
+        } => {
+            let device_context = find_device_for_user(&user_id)?;
+            let sdk = ironoxide::initialize(&device_context, &Default::default()).await?;
+            let g_id = GroupId::try_from(group_id.as_str())?;
+            let m_ids_or_error: std::result::Result<Vec<_>, _> = new_members
+                .iter()
+                .map(|u| UserId::try_from(u.as_str()))
+                .collect();
+            let add_result = sdk.group_add_members(&g_id, &m_ids_or_error?).await?;
+            let successes = add_result
+                .succeeded()
+                .iter()
+                .map(|user| user.id())
+                .collect::<Vec<_>>();
+            let failures = add_result
+                .failed()
+                .iter()
+                .map(|err| (err.user().id(), err.error()))
+                .collect::<Vec<_>>();
+            println!("Successful adds: {:?}", successes);
+            println!("Failed adds: {:?}", failures);
+        }
     }
 
     Ok(())
 }
 
+fn find_device_for_user(user_id: &str) -> Result<DeviceContext> {
```

Comment:
I think this maybe should just return an sdk instead.

Reply:
I agree